static/app/views/preprod/buildDetails/main/buildDetailsMetricCards.tsx

```diff
@@ -0,0 +1,218 @@
+  watchBreakdown?: WatchBreakdown;
+}
+
+interface WatchBreakdown {
+  appValue: string;
+  watchValue: string;
+}
+
+export function BuildDetailsMetricCards(props: BuildDetailsMetricCardsProps) {
+  const {
+    sizeInfo,
+    processedInsights,
+    totalSize,
+    platform: platformProp,
+    onOpenInsightsSidebar,
+  } = props;
+
+  if (!isSizeInfoCompleted(sizeInfo)) {
+    return null;
+  }
+
+  const labels = getLabels(platformProp ?? undefined);
+  const primarySizeMetric = getMainArtifactSizeMetric(sizeInfo);
+  const watchArtifactMetric = sizeInfo.size_metrics.find(
+    metric => metric.metrics_artifact_type === MetricsArtifactType.WATCH_ARTIFACT
+  );
+  const installMetricValue = formattedPrimaryMetricInstallSize(sizeInfo);
+  const downloadMetricValue = formattedPrimaryMetricDownloadSize(sizeInfo);
+  const totalPotentialSavings = processedInsights.reduce(
+    (sum, insight) => sum + (insight.totalSavings ?? 0),
+    0
+  );
+  const potentialSavingsPercentage =
+    totalSize > 0 ? totalPotentialSavings / totalSize : null;
+  const potentialSavingsPercentageText =
+    potentialSavingsPercentage !== null && potentialSavingsPercentage !== undefined
+      ? ` (${formatPercentage(potentialSavingsPercentage, 1, {
+          minimumValue: 0.001,
+        })})`
+      : undefined;
+
+  const metricsCards: MetricCardConfig[] = [
+    {
+      key: 'install',
+      title: labels.installSizeLabel,
+      icon: <IconCode size="sm" />,
+      labelTooltip: labels.installSizeDescription,
+      value: installMetricValue,
+      watchBreakdown: getWatchBreakdown(
+        primarySizeMetric,
+        watchArtifactMetric,
+        'install_size_bytes'
+      ),
+    },
+    {
+      key: 'download',
+      title: labels.downloadSizeLabel,
+      icon: <IconDownload size="sm" />,
+      labelTooltip: labels.downloadSizeDescription,
+      value: downloadMetricValue,
+      watchBreakdown: getWatchBreakdown(
+        primarySizeMetric,
+        watchArtifactMetric,
+        'download_size_bytes'
+      ),
+    },
+    {
+      key: 'savings',
+      title: t('Potential savings'),
+      icon: <IconLightning size="sm" />,
+      labelTooltip: t('Total savings from insights'),
+      value: formatBytesBase10(totalPotentialSavings),
+      percentageText: potentialSavingsPercentageText,
+      showInsightsButton: totalPotentialSavings > 0,
+    },
+  ];
+
+  return (
+    <Flex gap="lg" wrap="wrap">
+      {metricsCards.map(card => {
+        const valueContent = (
+          <Heading as="h3">
+            <MetricValue $interactive={Boolean(card.watchBreakdown)}>
+              {card.value}
+            </MetricValue>
+            {card.percentageText ?? ''}
+          </Heading>
+        );
+
+        return (
+          <MetricCard
+            key={card.key}
+            icon={card.icon}
+            label={card.title}
+            labelTooltip={card.labelTooltip}
+            action={
+              card.showInsightsButton
```

Comment:
Nit: I usually like to extract things like this out to a variable in a block above the `return()` statement. Something like:
```
let metricAction;
if (card.showInsightsButton) {
  metricAction = { ... }
}
```
then this just becomes
```
<MetricCard
     ...
     action={metricAction}
>
```